release/models/gnsi/openconfig-gnsi-certz.yang

```diff
@@ -0,0 +1,204 @@
+  import openconfig-system {
+    prefix oc-sys;
+  }
+  import openconfig-system-grpc {
+    prefix oc-sys-grpc;
+  }
+  import openconfig-types {
+    prefix oc-types;
+  }
+  import openconfig-yang-types {
+    prefix oc-yang;
+  }
+  import openconfig-gnsi {
+    prefix oc-gnsi;
+  }
+  organization
+    "OpenConfig Working Group";
+
+  contact
+    "OpenConfig working group
+    netopenconfig@googlegroups.com";
+
+  description
+    "This module provides a data model for the metadata of gRPC credentials
+    installed on a networking device.";
+
+  oc-ext:openconfig-version "0.5.0";
+
+  revision 2024-01-25 {
+    description
+      "Major style updates.";
+    reference "0.5.0";
+  }
+
+  revision 2023-08-24 {
+    description
+      "Adds ssl-profile-id leaf";
+    reference "0.4.0";
+  }
+
+  revision 2023-05-10 {
+    description
+      "Adds authentication policy freshness information.";
+    reference "0.3.0";
+  }
+
+  revision 2022-10-30 {
+    description
+      "Adds success/failure counters.";
+    reference "0.2.0";
+  }
+
+  revision 2022-09-20 {
+    description
+      "Initial revision.";
+    reference "0.1.0";
+  }
+
+  typedef version {
+    type string;
+    description
+      "The version ID of the credential as provided by the credential
+      manager when the credential was pushed. This leaf persists through
+      a reboot.";
+  }
+
+  typedef created-on {
+    type oc-types:timeticks64;
+    description
+      "The creation time of the credential as reported by the credential
+      manager when the credential was pushed to the device. This value is
+      reported as nanoseconds since epoch (January 1st, 1970 00:00:00 GMT).
+      This leaf persists through a reboot.";
+  }
+  // gRPC server related definitions.
+  // Success/failure counters.
+  grouping grpc-server-acctz-counters  {
+    description
+      "A collection of counters that were collected while evaluating
+      access to the gRPC server.";
+
+    container certz-counters {
+      config false;
+      description
+        "A collection of counters that were collected by the gRPC during
+        the authentication process.";
+
+      leaf access-rejects {
+        type oc-yang:counter64;
+        description
+          "The total number of times the gRPC denied access to the server.";
+      }
+      leaf last-access-reject {
+        type oc-types:timeticks64;
+        description
+          "A timestamp of the last time the gRPC denied access to
+          the server.";
+      }
+      leaf access-accepts {
+        type oc-yang:counter64;
+        description
+          "The total number of times the gPRC allowed access to
+          the server.";
+      }
+      leaf last-access-accept {
+        type oc-types:timeticks64;
+        description
+          "A timestamp of the last time the gRPC allowed access to
+          the server.";
+      }
+    }
+  }
+
+  /* TODO: move to a system-wide subtree:
```

Comment:
If this is going to be moved to a system-wide subtree, does it make sense to keep the counters here under grpc-server?
What about the counters for those GNSI SSL profiles which are NOT used by grpc-server?
If this counter is only meant for grpc-server and not widely applicable for other SSL profiles, then should it come out of certz? 

@earies Do you have any comments on these counters in certz?

Reply:
Please redirect this discussion to https://github.com/openconfig/public/issues/1049 and https://github.com/openconfig/public/issues/1050 for tracking this refactoring effort. I think it will speed up the process if we place this discussion in a separate issue.